contracts/src/deployers/aave/AaveHyperdriveDeployerCoordinator.sol

```diff
@@ -0,0 +1,206 @@
+///                    particular legal or regulatory significance.
+contract AaveHyperdriveDeployerCoordinator is HyperdriveDeployerCoordinator {
+    using FixedPointMath for uint256;
+    using SafeERC20 for ERC20;
+
+    /// @notice The deployer coordinator's kind.
+    string public constant override kind =
+        AAVE_HYPERDRIVE_DEPLOYER_COORDINATOR_KIND;
+
+    /// @notice Instantiates the deployer coordinator.
+    /// @param _name The deployer coordinator's name.
+    /// @param _factory The factory that this deployer will be registered with.
+    /// @param _coreDeployer The core deployer.
+    /// @param _target0Deployer The target0 deployer.
+    /// @param _target1Deployer The target1 deployer.
+    /// @param _target2Deployer The target2 deployer.
+    /// @param _target3Deployer The target3 deployer.
+    /// @param _target4Deployer The target4 deployer.
+    constructor(
+        string memory _name,
+        address _factory,
+        address _coreDeployer,
+        address _target0Deployer,
+        address _target1Deployer,
+        address _target2Deployer,
+        address _target3Deployer,
+        address _target4Deployer
+    )
+        HyperdriveDeployerCoordinator(
+            _name,
+            _factory,
+            _coreDeployer,
+            _target0Deployer,
+            _target1Deployer,
+            _target2Deployer,
+            _target3Deployer,
+            _target4Deployer
+        )
+    {}
+
+    /// @dev Prepares the coordinator for initialization by drawing funds from
+    ///      the LP, if necessary.
+    /// @param _hyperdrive The Hyperdrive instance that is being initialized.
+    /// @param _lp The LP that is initializing the pool.
+    /// @param _contribution The amount of capital to supply. The units of this
+    ///        quantity are either base or vault shares, depending on the value
+    ///        of `_options.asBase`.
+    /// @param _options The options that configure how the initialization is
+    ///        settled.
+    /// @return value The value that should be sent in the initialize transaction.
+    function _prepareInitialize(
+        IHyperdrive _hyperdrive,
+        address _lp,
+        uint256 _contribution,
+        IHyperdrive.Options memory _options
+    ) internal override returns (uint256) {
+        // If base is the deposit asset, the initialization will be paid in the
+        // base token.
+        address token;
+        address baseToken = _hyperdrive.baseToken();
+        if (_options.asBase) {
+            token = baseToken;
+        }
+        // Otherwise, the initialization will be paid in vault shares.
+        else {
+            // The token is the vault shares token.
+            token = _hyperdrive.vaultSharesToken();
+
+            // AToken transfers are in base, so we need to convert from
+            // shares to base.
+            _contribution = _convertToBase(baseToken, token, _contribution);
```

Comment:
i wanna make sure i understand this. 

shares - aDai
base - Dai

Let's say aDai = 2 Dai and i want to transfer 1 aDai does this mean that we are setting contribution = 2?

Reply:
Okay maybe i have this wrong and aTokens are the "Base" and scaledTokens (shudders) are the shares?

Reply:
okay so aDai is a rebasing token so this means: dai = aDai and shares is scalesTokens which is the units of contribution